pkg/tsdb/azuremonitor/macros/macros.go

```diff
@@ -34,8 +35,10 @@ type kqlMacroEngine struct {
 //   - $__escapeMulti('\\vm\eth0\Total','\\vm\eth2\Total') -> @'\\vm\eth0\Total',@'\\vm\eth2\Total'
 
 // KqlInterpolate interpolates macros for Kusto Query Language (KQL) queries
-func KqlInterpolate(query backend.DataQuery, dsInfo types.DatasourceInfo, kql string, defaultTimeField ...string) (string, error) {
-	engine := kqlMacroEngine{}
+func KqlInterpolate(logger log.Logger, query backend.DataQuery, dsInfo types.DatasourceInfo, kql string, defaultTimeField ...string) (string, error) {
+	engine := kqlMacroEngine{
+		logger: logger,
+	}
```

Comment:
not sure why you decided to store this in the struct vs passing it as an argument?

Reply:
Initially, I did not know how deep logger usage was but now it makes sense to change.